src/Components/Projects/Projects.tsx

```diff
@@ -138,25 +151,54 @@ const Projects: React.FC = () => {
         );
       });
   };
-
+  // **********search by proj name***********************
+  const getProjectTitleValue = (e: MouseEvent) => {
+   
+    getAllProjectsList(1, e.target.value);
+    setSearchString(e.target.value);
+   
+  };
+  // *****************************************************
   useEffect(() => {
-    getAllProjectsList();
+      getAllProjectsList(1,searchString);
   }, []);
+  // useEffect(() => {
+  //   const timerId = setTimeout(() => {
+  //     getAllProjectsList(1, searchString);
+  //   }, 500);
+  //   return () => clearTimeout(timerId);
+  // }, [searchString]);
+
   return (
     <>
       <div className="header d-flex justify-content-between p-3">
         <h3>projects</h3>
 
-        
-          <Link to={"/dashboard/add-project"} className="btn btn-warning rounded-5 px-4 customize-link">
-            <i className="fa fa-plus" aria-hidden="true"></i> &nbsp;Add New
-            Project
-          </Link>
-        
+        <Link
+          to={"/dashboard/add-project"}
```

Comment:
the add-project path needs to be nested after the projects  path to maintain a good URL hierarchy
/dashboard/projects/add-project